src/libmeasurement_kit/engine.cpp

```diff
@@ -133,10 +134,16 @@ nlohmann::json Task::wait_for_next_event() {
     if (!pimpl_->deque.empty()) {
         auto rv = std::move(pimpl_->deque.front());
         pimpl_->deque.pop_front();
-        return nlohmann::json{std::move(rv)};
+        return rv;
```

Comment:
I forgot to document this. It's a bugfix: previous code wrapped the event into a vector object.